src/tokio/caller.rs

```diff
@@ -0,0 +1,135 @@
+use crate::packet::PacketCodec;
+use crate::protocol::sender::{Sender, SenderAlgorithmAction};
+use crate::protocol::Handshake;
+use crate::{SrtCongestCtrl, ControlPacket, DataPacket, Packet};
+use bytes::Bytes;
+use failure::Error;
+use futures::channel::{mpsc, oneshot};
+use futures::future::Fuse;
+use futures::select;
+use futures::{FutureExt, SinkExt, StreamExt};
+use log::debug;
+use std::net::SocketAddr;
+use std::time::{Duration, Instant};
+use tokio::net::UdpSocket;
+use tokio::time::delay_until;
+use tokio_util::udp::UdpFramed;
+
+pub struct SrtCaller {
+    close_tx: Option<oneshot::Sender<()>>,
+    srt_send_tx: mpsc::Sender<(Bytes, Instant)>,
+    _task: tokio::task::JoinHandle<()>,
```

Comment:
Any reason that this is stored? Looking at the [docs](https://docs.rs/tokio/0.2.9/tokio/task/struct.JoinHandle.html) it seems like it detaches when it gets dropped, so if we're not joining on it dropping it seems fine.